File: spring-grpc-example/grpc-client-example/src/main/java/ru/vood/grpcdemo/client/LogGrpcInterceptor.java
package ru.vood.grpcdemo.client;

import io.grpc.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * @author Michael (yidongnan@gmail.com)
 * @since 2016/12/8
 */
public class LogGrpcInterceptor implements ClientInterceptor {

    private static final Logger log = LoggerFactory.getLogger(LogGrpcInterceptor.class);

    @Override
    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method,
                                                               CallOptions callOptions, Channel next) {
        log.info(method.getFullMethodName());
        return next.newCall(method, callOptions);
    }

}

File: spring-grpc-example/grpc-server-example/src/main/java/ru/vood/grpcdemo/server/interseptors/LogGrpcInterceptor.kt
package ru.vood.grpcdemo.server.interseptors

import io.grpc.Metadata
import io.grpc.ServerCall
import io.grpc.ServerCallHandler
import io.grpc.ServerInterceptor
import org.slf4j.LoggerFactory

class LogGrpcInterceptor : ServerInterceptor {
    override fun <ReqT, RespT> interceptCall(serverCall: ServerCall<ReqT, RespT>, metadata: Metadata,
                                             serverCallHandler: ServerCallHandler<ReqT, RespT>): ServerCall.Listener<ReqT> {
        log.info("===========================>" + serverCall.methodDescriptor.fullMethodName)
        return serverCallHandler.startCall(serverCall, metadata)
    }

    companion object {
        private val log = LoggerFactory.getLogger(LogGrpcInterceptor::class.java)
    }
}